Добавление плеера

NeonNik2245 opened the pull request:
Первоначальное добавление плеера с поддержкой выбора файла

NeonNik2245 committed:
Изменён внешний вид основной страницы

Добавлено разделение на колонки, протестировано размещение виджетов внутри колонок

NeonNik2245 committed:
Удаление лишней кнопки

NeonNik2245 committed:
Merge branch 'lch361:main' into main

NeonNik2245 committed:
Добавление видеоплеера

Добавлен видеоплеер, кнопки под него (весьма немного), разделение экрана под размер окна

lch361 commented on lib/main.dart:
```diff
@@ -57,6 +66,56 @@ class MyHomePage extends StatefulWidget {
+   );
+ 
+   if (result != null) {
+    String _videofilePath = result.files.single.path!;
```
Замечен бросок исключения. Может, добавим документацию к `getFileVideo()`, какое именно исключение может быть брошено в пределах этой функции?

lch361 commented on lib/main.dart:
```diff
@@ -31,13 +40,13 @@ class MyApp extends StatelessWidget {
         useMaterial3: true,
       ),
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+      home: const MyHomePage(title: 'Subtitle Editor Demo Home Page'), //Прямо тут можно задать новое имя, передаётся в строке 40
```
```suggestion
      home: const MyHomePage(title: 'Subtitle Editor Demo Home Page'), //Прямо тут можно задать новое имя, передаётся в [MyHomePage()]
```

lch361 commented on lib/main.dart:
```diff
@@ -57,6 +66,56 @@ class MyHomePage extends StatefulWidget {
+    String _videofilePath = result.files.single.path!;
+    print(_videofilePath);
+    player.open(Media(_videofilePath));
+     setState(() {});
```
Как я сразу не заметил - кажется, эта строка слегка вне стандартного формата кода. Рекомендую запустить `dart format` для кода тогда, или же можем предложить определённый стиль кода для нашего проекта.
```suggestion
    setState(() {});
```

lch361 commented on lib/main.dart:
```diff
@@ -68,15 +127,147 @@ class _MyHomePageState extends State<MyHomePage> {
+        // the App.build method, and use it to set our appbar title.
+        title: Text(widget.title),
+      ),
+      body: 
+      Row( //Тело, разделённое по колонкам
+      children: [
```
Ок, вот это точно нестандартное форматирование.
```suggestion
      body: Row( //Тело, разделённое по колонкам
        children: [
```

lch361 commented on lib/main.dart:
```diff
@@ -68,15 +127,147 @@ class _MyHomePageState extends State<MyHomePage> {
+                height: 120,
+                child: FloatingActionButton(
+                  onPressed: getFileVideo,
+                  tooltip: 'Choose video file',
+                  child: const Icon(Icons.video_call_rounded),
+                ),
+              ),
+
+              SizedBox(
+                width: 120,
+                height: 120,
+                child: FloatingActionButton(
+                  onPressed: getFileSubtitle,
+                  tooltip: 'Choose video file',
```
Так, стоп, вряд ли эти две кнопки одинаковы. Может быть...?
```suggestion
                  tooltip: 'Choose video file',
                  child: const Icon(Icons.video_call_rounded),
                ),
              ),

              SizedBox(
                width: 120,
                height: 120,
                child: FloatingActionButton(
                  onPressed: getFileSubtitle,
                  tooltip: 'Choose subtitle file',
```

lch361 commented on lib/main.dart:
```diff
@@ -68,15 +127,147 @@ class _MyHomePageState extends State<MyHomePage> {
+            
+          ),
+
+        ],),
```
Ещё одно чудо форматирования... `dart format` бы не позволил из ниоткуда стакать скобки на одной строке

lch361 commented on lib/main.dart:
```diff
@@ -68,15 +127,147 @@ class _MyHomePageState extends State<MyHomePage> {
+        ],),
+
+
+        //   child: Column(
+        //   // Column is also a layout widget. It takes a list of children and
+        //   // arranges them vertically. By default, it sizes itself to fit its
+        //   // children horizontally, and tries to be as tall as its parent.
+        //   //
+        //   // Column has various properties to control how it sizes itself and
+        //   // how it positions its children. Here we use mainAxisAlignment to
+        //   // center the children vertically; the main axis here is the vertical
+        //   // axis because Columns are vertical (the cross axis would be
+        //   // horizontal).
+        //   //
+        //   // TRY THIS: Invoke "debug painting" (choose the "Toggle Debug Paint"
+        //   // action in the IDE, or press "p" in the console), to see the
+        //   // wireframe for each widget.
+        //   //mainAxisAlignment: MainAxisAlignment.start,
+        //   children: <Widget>[
+        //     SizedBox(
+        //       width: MediaQuery.of(context).size.width,
+        //       height: MediaQuery.of(context).size.width * 9.0 / 16.0,
+        //       // Use [Video] widget to display video output.
+        //       child: Video(controller: controller),
+        //     ),
+        //     // Text(
+        //     //   '$_counter',
+        //     //   style: Theme.of(context).textTheme.headlineMedium,
+        //     // ),
+        //     // SizedBox(
+        //     //   width: 120,
+        //     //   height: 120,
+        //     //   child: FloatingActionButton(
+        //     //     onPressed: _incrementCounter,
+        //     //     tooltip: 'Increment',
+        //     //     child: const Icon(Icons.add),
+        //     //   ),
+        //     // ),
+        //   ],
+        // ),
```
Кажется, это комментирование кода, который был поставлен во Flutter по умолчанию. Так вот, так как у нас буквально есть Git, который сохраняет прошлое состояние нашего кода, мы можем смело удалять старые очерки.
```suggestion
```

lch361 commented on lib/main.dart:
```diff
@@ -86,7 +277,8 @@ class _MyHomePageState extends State<MyHomePage> {
         // the App.build method, and use it to set our appbar title.
         title: Text(widget.title),
       ),
-      body: Center(
+      body:
+      Center(
```
Ну чем тебе эта строка не понравилась цельной? :(

lch361 commented on lib/main.dart:
```diff
@@ -120,6 +312,6 @@ class _MyHomePageState extends State<MyHomePage> {
         child: const Icon(Icons.add),
       ), // This trailing comma makes auto-formatting nicer for build methods.
-    );
+    );*/
```
То же самое. Если это старый код из образца, зафиксированный у нас буквально в первом коммите, смело его сносим

lch361 commented on lib/main.dart:
```diff
@@ -68,15 +127,147 @@ class _MyHomePageState extends State<MyHomePage> {
+          SizedBox( // Коробка под видео
+            width: MediaQuery.sizeOf(context).width * 0.7,
+            //width: MediaQuery.of(context).size.width,
+            height: MediaQuery.of(context).size.width * 0.7 * 9.0 / 16.0,
```
Вполне возможно, можно выделить в программе константы типа `RATIO` вместо магических чисел, которые с первого чтения непонятно какую именно функцию выполняют.
```suggestion
            height: MediaQuery.of(context).size.width * 0.7 * RATIO,
```

И потом где-нибудь:
```dart
const RATIO = 9.0 / 16.0;
```

lch361 commented on lib/main.dart:
```diff
@@ -31,13 +40,13 @@ class MyApp extends StatelessWidget {
 
 class MyHomePage extends StatefulWidget {
-  const MyHomePage({super.key, required this.title});
+  const MyHomePage({super.key, required this.title}); //А здесь имя запрашивается, передаётся в строке 34
```
Комментарии, ссылающиеся на определённый номер строки - то есть, если мы в будущем напишем код поверх этих строк, все подобные комментарии потребуют изменения. Не легче ли тогда ссылаться на имена классов и конструкторов?
```suggestion
  const MyHomePage({super.key, required this.title}); //А здесь имя запрашивается, передаётся в поле [MaterialApp.home]
```

lch361 commented on lib/main.dart:
```diff
@@ -1,6 +1,15 @@
+
 void main() {
+  WidgetsFlutterBinding.ensureInitialized();
+  // Обязательная инициализации пакета media kit
```
```suggestion
  // Обязательная инициализация пакета media kit
```

NeonNik2245 committed:
Apply suggestions from code review

Co-authored-by: Lich <91026799+lch361@users.noreply.github.com>

NeonNik2245 committed:
Незначительные правки

NeonNik2245 committed:
Update main.dart

Изменение табуляций, порядка вызова ensureInitialized, добавлено отделение кнопок

NeonNik2245 committed:
dart format

NeonNik2245 committed:
Smoke test fix
